docs/networking/rancher-desktop-guest-agent.md

```diff
@@ -0,0 +1,149 @@
+# **[Rancher Desktop Guest Agent](https://github.com/rancher-sandbox/rancher-desktop-agent)**
+
+The Rancher Desktop Guest Agent runs in the Rancher Desktop WSL distro, it runs inside the isolated namespace (when network tunnel is enabled). It is currently responsible for interaction between various APIs for container engines such as Moby, containerd, and Kubernetes. It listens for container creation events from those APIs if it detects any ports that require exposing from those containers it forwards the corresponding port mappings to the internal services.
+
+The Rancher Desktop Guest Agent operates within the Rancher Desktop WSL distribution, particularly in an isolated namespace when the network tunnel is enabled. It facilitates interactions between various container engine APIs like Moby, containerd, and Kubernetes. The agent monitors container/service creation events from these APIs and, upon detecting ports needing exposure, forwards the port mappings to internal services accordingly. This ensures efficient and automated port forwarding management within the Rancher Desktop environment.
+
+```mermaid
+flowchart  LR;
+subgraph Host["HOST"]
+host-switch["host-switch\n/services/forwarder/all\n /services/forwarder/expose\n /services/forwarder/unexpose"]
+end
+ subgraph VM["WSL"]
+  subgraph netNs["Isolated Network Namespace"]
+   guest-agent["Guest Agent"]
+   docker(("Docker API"))
+   containerd(("Containerd API"))
+   kubernetes(("K8s API"))
+   iptables(("iptable scanning"))
+   guest-agent <----> docker
+   guest-agent <----> containerd
+   guest-agent <----> kubernetes
+   guest-agent <----> iptables
+   guest-agent ----> host-switch
+  end
+  subgraph defaultNs["Default Namespace"]
+  wsl-proxy["wsl-proxy"]
+  end
+  guest-agent ----> |UNIX socket| wsl-proxy
+ end
+```
+
+### Supported Flags
+
+-   **debug**: Enables debug logging.
+
+-   **docker**: When this flag is enabled, the guest agent monitors Docker API container events via the Docker socket (`/var/run/docker.sock`). On container creation or deletion, if there are exposed ports, the agent creates corresponding port mappings. These mappings are forwarded to the Rancher Desktop Networking's `host-switch`, which hosts an API for exposing ports from the host into the network namespace. If WSL integration is enabled in Rancher Desktop, the port mapping is also forwarded to Rancher Desktop Networking’s `wsl-proxy`, allowing access to the container's ports from other WSL distributions.
```

Comment:
I think it would be better phrase like:

> When this flag is enabled, port mapping via docker API monitoring is enabled. See the [port mapping](#PortMapping) and [Docker](#Docker) sections below for details.